core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -228,6 +235,54 @@ func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
 
 	logger.Debugw(fmt.Sprintf("EthConfirmer: fetching receipts for %v transaction attempts", len(attempts)), "blockNum", blockNum)
 
+	attemptsByAddress := make(map[gethCommon.Address][]models.EthTxAttempt)
+	for _, att := range attempts {
+		attemptsByAddress[att.EthTx.FromAddress] = append(attemptsByAddress[att.EthTx.FromAddress], att)
+	}
+
+	for from, attempts := range attemptsByAddress {
+		latestBlockNonce, err := ec.getNonceForLatestBlock(ctx, from)
+		logger.Debugw(fmt.Sprintf("EthConfirmer: There are %v attempts from address %v, and latest nonce is %v",
+			len(attempts), from, latestBlockNonce), "blockNum", blockNum)
+
+		if err != nil {
+			return errors.Wrapf(err, "unable to fetch pending nonce for address: %v", from)
+		}
+
+		likelyConfirmed := make([]models.EthTxAttempt, 0)
+		for i := 0; i < len(attempts); i++ {
+			if attempts[i].EthTx.Nonce != nil && *attempts[i].EthTx.Nonce >= int64(latestBlockNonce) {
+				likelyConfirmed = attempts[0:i]
+				attempts = attempts[i:]
+				break
+			}
+		}
+
+		err = ec.fetchAndSaveReceipts(ctx, likelyConfirmed, blockNum)
+		if err != nil {
+			return errors.Wrapf(err, "unable to fetch and save receipts for likely confirmed txs, for address: %v", from)
+		}
+
+		err = ec.fetchAndSaveReceipts(ctx, attempts, blockNum)
```

Comment:
Hm, does it actually save us anything to do it again here?

Reply:
hm that was suppose to be the approach: "The first call would figure out which receipts to look up, and then we could batch the lookup of other receipts in a second call." 

so i assumed that the savings are about having *some* transactions confirmed earlier than others